conjur-api/Variable.cs

```diff
@@ -47,17 +48,39 @@ public string GetValue()
         /// </summary>
         /// <param name="val">Secret value.</param>
         public void AddSecret(string val)
+        {
+            byte[] data = Encoding.UTF8.GetBytes(val);
+            AddSecret(data);
+        }
+
+        /// <summary>Set a secret (value) to this variable.</summary>
+        /// <remarks>
+        /// The value is being cleared after being used!!!
+        /// The clearing is done so no trace of the secret will be available in the dump.
+        /// </remarks>
+        /// <param name="val">Secret value as byte array.</param>
+        public void AddSecret(byte[] val)
         {
             WebRequest webRequest = this.Client.AuthenticatedRequest(this.path);
             webRequest.Method = "POST";
+            if (webRequest is HttpWebRequest)
+            {
+                (webRequest as HttpWebRequest).AllowWriteStreamBuffering = false;
+            }
 
-            byte[] data = Encoding.UTF8.GetBytes(val);
             webRequest.ContentType = "text\\plain";
-            webRequest.ContentLength = data.Length;
-            webRequest.GetRequestStream().Write(data, 0, data.Length);
-            using (webRequest.GetResponse())
+            webRequest.ContentLength = val.Length;
+            using (Stream requestStream = webRequest.GetRequestStream())
+            {
+                requestStream.Write(val, 0, val.Length);
+                using (webRequest.GetResponse())
+                {
+                    // Intentional do not care about response content
+                }
+            }
+            for (int index = 0; index < val.Length; index++)
```

Comment:
Can we use:
Array.Clear(myArray, 0, myArray.Length);

https://docs.microsoft.com/en-us/dotnet/api/system.array.clear?redirectedfrom=MSDN&view=netframework-4.7.2#System_Array_Clear_System_Array_System_Int32_System_Int32_

And keep in mind that as long as there is a reference, no GC will occur

Reply:
refer and accept all comments